### PR TITLE
Secure headers for HTTP Trigger

### PR DESCRIPTION
**Pull request by maganaluis**
Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

This solves issue https://github.com/argoproj/argo-events/issues/1227 and it adds support for adding headers to the HTTP Trigger as secrets, so tokens are not committed as code. See the example below: 

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Sensor
metadata:
  name: http-trigger
spec:
  dependencies:
    - name: test-dep
      eventSourceName: webhook
      eventName: example
  triggers:
    - template:
        name: http-trigger
        http:
          secureHeaders:
            - name: serviceToken
              valueFrom:
                secretKeyRef:
                  name: dev-services-secret
                  key: serviceToken
          url: http://http-server.default.svc.cluster.local:8080/hello
          method: GET
      retryStrategy:
        steps: 3
        duration: 3s
      policy:
        status:
          allow:
            - 200
            - 201
```

### CODE REVIEW

File: pkg/apis/common/common.go [hunk 1]
```diff
@@ -108,6 +108,11 @@ type BasicAuth struct {
 	Password *corev1.SecretKeySelector `json:"password,omitempty" protobuf:"bytes,2,opt,name=password"`
 }
 
+type SecureHeaders struct {
+	HeaderName string                    `json:"headerName,omitempty" protobuf:"bytes,1,opt,name=headerName"`
+	Value      *corev1.SecretKeySelector `json:"value,omitempty" protobuf:"bytes,2,opt,name=value"`
```
[the remark this answers is not in the document]
Comment (maganaluis) [reply pkg/apis/common/common.go:113]: Are you saying that header names should be read from configmaps and values from secrets? What would be the benefit from reading from configmaps? @whynowy

[the remark this answers is not in the document]
Comment (whynowy) [reply pkg/apis/common/common.go:113]: No. I mean the value could come from configmaps OR secrets - similar to `envFrom`.

I think this struct could also be reused by the `metadata` carried by each of the event source, this is another topic.

**Comment by maganaluis** [comment]
@whynowy I tested with both secrets and configmaps. Given you switch to configMapKeyRef.

e.g. 

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: dev-services-secret
  namespace: argo-events
data:
  serviceToken: secret
```

**Review by whynowy — APPROVED** [review]
I give my conditional approval, please return `make codegen`, also an example would be appreciated. Thanks for your contribution!

**Comment by maganaluis** [comment]
@whynowy I re-ran it with the latest Kustomize version and looks like that worked. It still generated event-bus.md and eventsource.md, though I discarded those changes. Let me know if this is good, I don't have access to re-run test cases.

**Comment by whynowy** [comment]
> @whynowy I re-ran it with the latest Kustomize version and looks like that worked. It still generated event-bus.md and eventsource.md, though I discarded those changes. Let me know if this is good, I don't have access to re-run test cases.

No worries, I'll handle that.

**Comment by Hunter-Thompson** [comment]
@whynowy, any ETA on the next release? Would love this feature!

**Comment by whynowy** [comment]
> @whynowy, any ETA on the next release? Would love this feature!

I would expect the BitBucket implementation PR being merged before cutting v1.4, if that doesn't happen, I will tag the version anyways by the end of next week.

**Pull request merged**